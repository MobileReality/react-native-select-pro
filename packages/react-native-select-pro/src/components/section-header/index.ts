export { SectionHeader } from './section-header';
