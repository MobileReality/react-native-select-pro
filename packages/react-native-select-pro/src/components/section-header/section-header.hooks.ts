import { useOptionsListContext } from '../../context';

export const useSectionHeader = () => {
    const {
        styles,
        onPressSection,
        sectionHeaderButtonProps,
        sectionHeaderTextProps,
        sectionHeaderImageProps,
        multiple,
        disabled,
    } = useOptionsListContext();

    const { sectionHeader: sectionHeaderCustomStyles } = styles ?? {};

    const isDisabled = disabled ?? !multiple;

    return {
        onPressSection,
        sectionHeaderButtonProps,
        sectionHeaderTextProps,
        sectionHeaderImageProps,
        multiple,
        disabled,
        sectionHeaderCustomStyles,
        isDisabled,
    };
};
