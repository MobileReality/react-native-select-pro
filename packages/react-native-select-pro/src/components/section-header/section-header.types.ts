export type SectionHeaderProps = {
    title: string;
    isSelected: boolean;
};
