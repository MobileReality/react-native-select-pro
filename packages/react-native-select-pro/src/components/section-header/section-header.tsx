import React from 'react';
import type { ImageStyle, TextStyle, ViewStyle } from 'react-native';
import { Image, Pressable, StyleSheet, Text } from 'react-native';

import { COLORS, FONT_SIZE, PADDING, PRESSED_STYLE } from '../../constants';

import { useSectionHeader } from './section-header.hooks';
import type { SectionHeaderProps } from './section-header.types';

const iconSource = require('./../../assets/icons/x.png');

export const SectionHeader = ({ title, isSelected }: SectionHeaderProps) => {
    const {
        sectionHeaderCustomStyles,
        onPressSection,
        sectionHeaderButtonProps,
        sectionHeaderTextProps,
        sectionHeaderImageProps,
        isDisabled,
    } = useSectionHeader();

    return (
        <Pressable
            disabled={isDisabled}
            accessibilityState={{ disabled: isDisabled }}
            {...sectionHeaderButtonProps}
            style={({ pressed }) => [
                styles.sectionHeaderContainerStyle,
                sectionHeaderCustomStyles?.container,
                isSelected && sectionHeaderCustomStyles?.selected?.container,
                pressed && (sectionHeaderCustomStyles?.pressed ?? PRESSED_STYLE),
            ]}
            onPress={() => onPressSection(title)}
        >
            <Text
                {...sectionHeaderTextProps}
                style={[
                    styles.sectionHeaderTextStyle,
                    sectionHeaderCustomStyles?.text,
                    isSelected && sectionHeaderCustomStyles?.selected?.text,
                ]}
            >
                {title}
            </Text>
            {isSelected && (
                <Image
                    source={iconSource}
                    {...sectionHeaderImageProps}
                    style={[styles.xIcon, sectionHeaderCustomStyles?.clear?.icon]}
                />
            )}
        </Pressable>
    );
};

type Styles = {
    sectionHeaderContainerStyle: ViewStyle;
    sectionHeaderTextStyle: TextStyle;
    xIcon: ImageStyle;
};

const styles = StyleSheet.create<Styles>({
    sectionHeaderTextStyle: {
        fontSize: FONT_SIZE,
        color: COLORS.BLACK,
        fontWeight: 'bold',
        textAlign: 'left',
    },
    sectionHeaderContainerStyle: {
        padding: PADDING,
        backgroundColor: COLORS.WHITE,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    xIcon: {
        width: 16,
        height: 16,
    },
});
