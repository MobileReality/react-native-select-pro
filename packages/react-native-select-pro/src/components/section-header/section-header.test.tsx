import React from 'react';
import { render } from '@testing-library/react-native';

import { SectionHeader } from './section-header';

describe('SectionHeader', () => {
    it('should generate SectionHeader snapshot', () => {
        const sectionHeader = render(
            <SectionHeader
                title="Test Section"
                sectionHeaderTextStyle={undefined}
                sectionHeaderContainerStyle={undefined}
            />,
        );
        expect(sectionHeader).toMatchSnapshot();
    });
});
