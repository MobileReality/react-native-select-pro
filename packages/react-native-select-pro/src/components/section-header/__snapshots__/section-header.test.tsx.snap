// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader should generate SectionHeader snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "padding": 12,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 12,
          "fontWeight": "bold",
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    Test Section
  </Text>
</View>
`;
